src/sudo/pipeline.rs

```diff
@@ -42,18 +42,15 @@ pub struct Pipeline<Policy: PolicyPlugin, Auth: AuthPlugin> {
 }
 
 impl<Policy: PolicyPlugin, Auth: AuthPlugin> Pipeline<Policy, Auth> {
-    pub fn run(mut self, cmd_opts: SudoRunOptions) -> Result<(), Error> {
-        if !cmd_opts.env_var_list.is_empty() {
-            eprintln_ignore_io_error!(
-                "warning: CLI-level env var list has not yet been implemented and will be ignored"
-            )
-        }
+    pub fn run(mut self, mut cmd_opts: SudoRunOptions) -> Result<(), Error> {
         if !cmd_opts.preserve_env.is_nothing() {
             eprintln_ignore_io_error!(
                 "warning: `--preserve-env` has not yet been implemented and will be ignored"
             )
         }
 
+        let user_requested_env_vars = std::mem::take(&mut cmd_opts.env_var_list);
+
         let pre = self.policy.init()?;
         let mut context = build_context(cmd_opts.into(), &pre)?;
```

Comment:
Maybe having a method to split `SudoRunOptions` into `OptionsForContext` and `env_var_list` would be nicer? That would make it clear that the `mem::take` above does not silently cause `build_context` to operate on an empty `env_var_list`.